### ISSUE TITLE
Relative paths in `options.bootScripts`

### ISSUE THREAD

**Issue by bajtos** [strongloop/loopback-boot#65]
#31 introduced `options.bootScripts` that can be used to specify additional boot scripts to execute. The script paths are passed directly to `require`, which means that relative paths are resolved relatively to `node_modules/loopback-boot/lib`.

We should improve the compiler and resolve any relative paths using `appRootDir`:

```
/absolute/path.js -> /absolute/path/.js
module/relative.js -> module/relative.js
./app/relative -> {appRootDir}/app/relative.js
```

A similar problem in the way how modules, middleware and components are resolved is tracked by #73.

**Comment by bajtos** [comment]
@PradnyaBaviskar I believe the problem should be reproducible by modifying an existing unit-test in [compiler.test.js](https://github.com/strongloop/loopback-boot/blob/bfcd3ff15e35293b4c4497fc3d01c6ed164d68b5/test/compiler.test.js#L397-L406).

```js
    it('supports `bootScripts` option', function() {
      appdir.createConfigFilesSync();
      var initJs = appdir.writeFileSync('custom-boot/init.js',
        'module.exports = function(app) { app.fnCalled = true; };');
      var instructions = boot.compile({
        appRootDir: appdir.PATH,
        bootScripts: ['./custom-boot/init.js'] // << MODIFIED
      });
      expect(instructions.files.boot).to.eql([initJs]);
    });
```

**Comment by bajtos** [comment]
Fixed by #104. Note that module paths are still resolved against loopback-boot node_modules, see #73.
